hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ModifyTableProcedure.java

```diff
@@ -125,7 +185,11 @@ protected Flow executeFromState(final MasterProcedureEnv env, final ModifyTableS
           break;
         case MODIFY_TABLE_PRE_OPERATION:
           preModify(env, state);
-          setNextState(ModifyTableState.MODIFY_TABLE_CLOSE_EXCESS_REPLICAS);
+          if (reopenRegions) {
+            setNextState(ModifyTableState.MODIFY_TABLE_CLOSE_EXCESS_REPLICAS);
```

Comment:
I think we need a comment somewhere, probably here and in the validation code above, stating that it's important we don't allow region replica changes with reopenRegions=false because that mode skips state management necessary for region replica changes.

Reply:
There are two states in the main procedure process that skip handling replica-related management when 'reopenRegions' is set to 'false'. I've added a comment in the first state.